exe/facter.cc

```diff
@@ -299,6 +281,9 @@ int main(int argc, char **argv)
             log(level::warning, "Facter was called recursively, skipping external facts. Add '--no-external-facts' to silence this warning");
           } else {
             environment::set("INSIDE_FACTER", "true");
+            if (vm.count("external-dir")) {
+                external_directories.push_back(vm["external-dir"].as<string>());
```

Comment:
I have been unable to get it to properly parse any kind of list, no matter what formatting of cli arg I used. Since the description says the user is only supposed to be able to specify one directory, was there any reason to have it grab a vector other than to allow it to automatically save into `external_directories`? I had to refactor this functionality when I pulled out the config file code (to avoid requiring array references to be passed into the utility functions), but I think it would work just as well to use `vector<string>` as `string`.  Would doing the assignment be better here?